js/formValidator.js

```diff
@@ -0,0 +1,37 @@
+const form = document.getElementById('form');
+const email = document.getElementById('email');
+const name = document.getElementById('name');
+const message = document.getElementById('message');
+const formError = document.querySelector('.form-error');
+
+function showError() {
+  if (email.validity.valueMissing) {
+    formError.innerHTML = 'You need to enter an e-mail address.';
+  } else if (email.validity.typeMismatch) {
+    formError.innerHTML = 'Entered value needs to be an e-mail address.';
+  } else if (email.validity.patternMismatch) {
+    formError.innerHTML = 'Entered value needs to be lowercase characters';
+  }
+  formError.className = 'form-error';
+}
+
+email.addEventListener('input', () => {
+  if (email.validity.valid) {
+    formError.innerHTML = '';
+    formError.className = '';
+  } else {
+    showError();
+  }
+});
+
+form.addEventListener('submit', (event) => {
+  if (name.validity.valueMissing || message.validity.valueMissing || email.validity.valueMissing) {
+    formError.innerHTML = 'You need to fill all the fields.';
+    formError.className = 'form-error';
+    event.preventDefault();
+  }
```

Comment:
You should not use name property, it is no longer recommended to rely on the "name" property to determine the name variable. I recommend you to use the other name variable. 
for more [info](https://stackoverflow.com/questions/65379554/why-does-my-variable-show-its-deprecated)
![image](https://github.com/waiyankyaw961999/Microverse-portofolio/assets/43877689/1fa8fe22-3f8e-4528-be4b-fe654e74bf80)